src/Microsoft.Tye.Hosting/Dashboard/Shared/MetricsDisplay.razor

```diff
@@ -0,0 +1,74 @@
+@implements IDisposable
+@using System.Threading;
+
+<h4>Metrics</h4>
+<ReplicaSelector Service="@Service" ReplicaChanged="OnReplicaChanged" />
+
+<div class="px-4 py-2">
+<table class="table table-striped table-bordered">
+    <colgroup>
+       <col span="1" style="width: 75%;">
+       <col span="1" style="width: 25%;">
+    </colgroup>
+    <thead class="thead-light">
+        <tr>
+            <th scope="col">Metric</th>
+            <th scope="col">Value</th>
+        </tr>
+    </thead>
+    @if (replica is object)
+    {
+        <tbody>
+            @foreach (var metric in replica.Metrics.OrderBy(kvp => kvp.Key))
+            {
+                <tr>
+                    <td>@metric.Key</td>
+                    <td>@metric.Value</td>
+                </tr>
+            }
+        </tbody>
+    }
+    else
+    {
+        <tbody></tbody>
+    }
+</table>
+</div>
+
+@code {
+    ReplicaStatus? replica;
+    Timer? timer;
+
+    [Parameter] public Service Service { get; set; } = default!;
+
+    protected override void OnParametersSet()
+    {
+        replica = Service?.Replicas.FirstOrDefault().Value;
+        OnReplicaChanged(replica);
+    }
+
+    void OnReplicaChanged(ReplicaStatus? replica)
+    {
+        this.replica = replica;
+        if (this.replica == null && timer is object)
+        {
+            timer.Dispose();
```

Comment:
this would be cleaner with an observable, but it would require some new plumbing and probably use of immutabledictionary (or at least immutable snapshots as a pattern).

Reply:
Instead of that can we move this logic elsewhere and hide it behind an IObservable/Subject?

Reply:
I did most of this but ran into a problem with only has solutions I hate...

The problem is that event counters fire once per-counter per-tick (the counters are not batched. 

So the solution could be to use immutable snapshots (`ImmutableDictionary<>`, and then fire once each time each counter updates. Then you'd want an observable that does buffering in between the producer and consumer. This would have more moving parts and use more resources than what we're doing.

If we scale up the number of consumers of the metrics then it makes sense to move to a push-based model.

Or if you know of a way to have eventcounters do batching 😆